src/app/services/server-authentication.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { ServerApiService } from './server-api.service';
+import { Observable } from 'rxjs';
+import { Response } from '../../../api-objects/GenericResponse'
+import { LoginSession } from 'api-objects/LoginSession';
+import { tap } from 'rxjs/operators';
+import { Utilities } from '../common/utlilities';
+
+
+@Injectable()
+export class ServerAuthenticationService {
+    //constructor(private serverApiService: ServerApiService) { }
+    constructor(private serverApiService: ServerApiService) { }
+
+    public login(username: string, password: string): Observable<Response<LoginSession>> {
+        return this.serverApiService.post<LoginSession>('login', { 'username': username, 'password': password })
+            //TODO: Find technique of piping authentication into session storrage  
+            .pipe(tap((response) => {
+                Utilities.setSessionStorage('userId', response.data.user.toString());
+                Utilities.setSessionStorage('sessionId', response.data.session.toString());
+                // Making Permission Check
+                if (response.data.authorityType > 0) {
+                    Utilities.setSessionStorage('permissions', 'admin');
+                } else {
+                    Utilities.setSessionStorage('permissions', 'audiologist');
```

Comment:
In my pull request, api-objects/UsersObject.ts contains:
export const authorityTypes: Array<string> = ['Admin', 'Audiologist'];
If you write: Utilities.setSessionStorage('permissions', response.data.authorityType);
Then we just need to change audiologist-navigation/navigation-aids.ts determineState() to also use it, and no more magic strings for permissions.
Of course, the backend will need to use it as well to ensure that don't mix up which is which.  But then it would be easier to add more User types.